src/kps/business/RouteCalculator.java

```diff
@@ -10,12 +10,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class RouteCalculator {
     private List<Route> routes;
 
     public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getRoutes();
+        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
```

Comment:
I changed the line inside getUniqueRoutes if(!r1.getDate().after(r2.getDate())) to if(r1.getDate().before(r2.getDate())) and it works. I guess it is because most of the events have the exact same date.